share/steel/examples/pulse/lib/Pulse.Lib.LinkedList.fst

```diff
@@ -124,27 +121,15 @@ fn cases_of_is_list (#t:Type) (x:llist t) (l:list t)
         Cons head tl -> { 
             rewrite (is_list x l) as (is_list x (head::tl));
             elim_is_list_cons x head tl;
-            assert pure (Some? x);
-            match x {
-                None -> {  
-                    elim_false (is_list_cases x l);
-                    with tail. assert(is_list tail tl);
-                    drop (is_list tail tl);
-                    with v. assert (pts_to v (mk_node head tail));
-                    drop (pts_to v (mk_node head tail));
-                }
-                Some v -> { 
-                    with tail. assert (is_list #t tail tl);
-                    with w. assert (pts_to w (mk_node head tail));
-                    rewrite (pts_to w (mk_node head tail)) as
-                            (pts_to v (mk_node head tail));
-                    rewrite (is_list tail tl) as 
-                            (is_list (mk_node head tail).tail tl);
-                    fold (is_list_cases (Some v) l);
-                    rewrite (is_list_cases (Some v) l) as
-                            (is_list_cases x l)
-                }
-            }
+            assert pure (Some? x); //surprising that this is needed
+            let v = Some?.v x;
```

Comment:
Similar bit of SMT strangeness here: needed to assert before projecting ... the assert should not be needed.